### ISSUE TITLE
Mightier Amp app seems not able to connect to NUX Mighty Plug after upgrading Mighty Plug to v2.1 firmware

### ISSUE THREAD

**Issue by rspiezia** [tuntorius/mightier_amp#10]
Hi tuntorius,
after the release of V2.1 firmware of Mighty Plug by NUX your app seems not able to connect to NUX for pre-set management.
Probably NUX has changed something in the "communication protocol" with the hardware in this new firmware release.
Try to check it.

Anyway thanks a lot  for your amazing app

Regards

**Comment by tuntorius** [comment]
Actually the problem is very simple. The've increased the EFX count from 13 to 14 which my app doesn't expect. If that 14th effect (Rose Comp) is used in one of the 7 presets, my amp tries to reference an effect it doesn't know about and produces an error. As a temporary fix you can use Mighty Amp to make sure none of the 7 presets use Rose Comp. Bear in mind that Mightier amp will show incorrect amps and fx because they've rearranged them.

I'm working on the new version and it will be ready in a few days.

**Comment by rspiezia** [comment]
Thank you for your quick feedback.
Yes I've seen that they have re-arranged "amps&effects" section and also the app is been improved but yours is definitely better.
Please also note that also the pre-set area in the new app now is "free". 
It means you can choose everything you want for all the 7 pre-set slots you have.
You are not obliged to use green for clean, orange for "crunch" and etc... and switch between guitar & bass
Now you can use the 7 slots as you prefer (for sure they found inspiration from your app for this).
Please also note that they added some new controls in some amps simulation

Looking forward for the new version
Regards

**Comment by itarozzi** [comment]
O.T. :

Hi Tuntorius, as a developer I'm just curious: do you have any contact with NUX developers or is your work all about reverse engineering?

With the success of your app, I think they should at least open the specs to make your job easier.  If not, I think we, as user, should press for that

**Comment by T40ma5** [comment]
Same issue with Mighty Air. I had not used Mightier Amp prior but I'm on firmware v2.1 and it will not connect.

**Comment by Djr53** [comment]
I'm also unable to connect after the upgrade to firmware v2.1. The app has been excellent and I'm looking forward to the new version.

**Comment by tuntorius** [comment]
Did you update to the latest version? I released v 0.8.6 few days ago. Avoid 0.8.5. It was available for a brief period of time but had a bug that prevented any kind of connection - both with old and new fw version.

**Comment by T40ma5** [comment]
Using v 0.8.6 and all is well. It is working. However, as a note, I had to uninstall and install the app to get to v 0.8.6. Version 0.8.2 was not giving the option to install the newer version, it was just reporting installed in the play store.
On Sep 8 I had installed app version 0.8.2 from google play (the day I posted the initial problem). Not sure when v 0.8.5 was made available but must've just missed it. It's possible that earlier versions are not indicating to the play store to install the update?
